ports/rp2/boards/W5100S_EVB_PICO/mpconfigboard.cmake

```diff
@@ -1,4 +1,3 @@
 # cmake file for Wiznet W5100S-EVB-Pico.
 set(PICO_BOARD wiznet_w5100s_evb_pico)
 set(MICROPY_PY_NETWORK_WIZNET5K W5100S)
-set(MICROPY_PY_LWIP 1)
```

Comment:
These should stay (unless there's some reason we want to stop using LWIP on these boards?) @andrewleech ?

Reply:
First, the chip of WIZNET must run its own hardware TCP/IP by default.
Working with LWIP should be an option.

Therefore, if a value of 1 is given to MICROPY_PY_LWIP in the mpconfigboard.cmake , even if the LWIP(=MICROPY_PY_LWIP) value is set to '0' as an option when running make, the MICROPY_PY_LWIP value is eventually changed to '1' by the value of mpconfigboard.cmake.

For reference, when operating in LWIP, Ethernet operation speed is slow.

Reply:
> First, the chip of WIZNET must run its own hardware TCP/IP by default.

Why? Is it just the speed difference? How much difference does it make?

From MicroPython's perspective, for consistency, it's better if we use the same stack on all boards and ports.

Reply:
I just did some performance tests.

With the SPI baudrate at 20M:
  LWIP: 5.5Mbps.
  Hardware TCP: 10.5Mbps

At 10M:
  LWIP: 3.7Mbps
  Hardware TCP: 6.6Mbps

So about twice as fast. @irinakim12 why is this? Is it a hardware limitation or a driver issue? I'm curious why the SPI baudrate affects it like this, is there a (very!) significant per-frame overhead in the raw mode?

@andrewleech Do you know more about this?

Reply:
When I was testing the initial integration I did find the hardware TCP stack to be faster as found, but somewhat more limited - lwip can have any number of sockets (only limited by ram) and a "full" range of other network features. The hardware TCP bindings were missing a few features, though I think I added a couple of them to satisfy unit tests... can't remember for sure now.

I never really investigated the speed difference of the two stacks, though some of it was due / affected by the background polling of lwip. I'd attempted to improve on that with the interrupts, though didn't thoroughly profile the remaining speed differences.

SPI clock affecting speed appears to be expected though: https://docs.wiznet.io/Product/iEthernet/W5500/Application/spi-performance#data-throughput-according-to-spi-clock-speed

Reply:
@jimmo  I also tried to modify the code because the speed slowed down when operating with LWIP, but I don't know where the lock takes place.  Since it is a hardware device, it is unavoidably limited, but I understand that 8 channels are available, and the current code is coded to use only 1 channel.